kv/kvserver: TestStoreRangeMergeDuringShutdown failed

kv/kvserver.TestStoreRangeMergeDuringShutdown [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/13700981?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/13700981?buildTab=artifacts#/) on release-23.1 @ [19ba39bf5c28cd15b89796332c0bbf6d4a7b0f50](https://github.com/cockroachdb/cockroach/commits/19ba39bf5c28cd15b89796332c0bbf6d4a7b0f50):


```
=== RUN   TestStoreRangeMergeDuringShutdown
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/c6156e329ef0e4dd2f31712f19cdbe2a/logTestStoreRangeMergeDuringShutdown129516135
    test_log_scope.go:79: use -show-logs to present logs inline
    client_merge_test.go:4193: expected "not lease holder" error, but got <nil>
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/c6156e329ef0e4dd2f31712f19cdbe2a/logTestStoreRangeMergeDuringShutdown129516135
--- FAIL: TestStoreRangeMergeDuringShutdown (1.70s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeDuringShutdown.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35688

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -4133,7 +4133,7 @@ func TestStoreRangeMergeDuringShutdown(t *testing.T) {
 			// Sleep to give the shutdown time to propagate. The test appeared to work
 			// without this sleep, but best to be somewhat robust to different
 			// goroutine schedules.
-			time.Sleep(10 * time.Millisecond)
+			time.Sleep(20 * time.Millisecond)
 		} else {
 			state.Unlock()
 		}

